### PR TITLE
Add conda recipe for bmi-fortran

### PR DESCRIPTION
**Pull request by mdpiper**
This PR adds a conda recipe for the `bmi-fortran` package, which defines the Fortran 2003 bindings for the [Basic Model Interface](https://bmi.readthedocs.io).

Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages
- [x] Build number is 0
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by mdpiper** [comment]
@conda-forge/staged-recipes This PR is ready to go. Please let me know if I can make any improvements!

File: recipes/bmi-fortran/build.sh [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#! /usr/bin/env bash
+
+mkdir _build && cd _build
+
+cmake .. \
+  -DCMAKE_INSTALL_PREFIX=$PREFIX \
+  -DCMAKE_BUILD_TYPE=Release
+
+make
```
Comment (chrisburr) [inline recipes/bmi-fortran/build.sh]:
```suggestion
make -j${CPU_COUNT}
```

File: recipes/bmi-fortran/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  sha256: 2ade92d7bfa494d379fa6fea4d03734cfd640d46840b944ce013a34a933c5ef5
+
+build:
+  number: 0
+  skip: True  # [win]
+
+requirements:
+  build:
+    - cmake
+    - {{ compiler('fortran') }}
```
Comment (chrisburr) [inline recipes/bmi-fortran/meta.yaml]:
```suggestion
    - {{ compiler('fortran') }}
    - make
```

File: recipes/bmi-fortran/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+test:
+  commands:
+    - test -f $PREFIX/include/bmif_{{ version|replace(".","_") }}.mod
+    - test -h $PREFIX/lib/libbmif$SHLIB_EXT
+
+about:
+  home: https://bmi.readthedocs.io
+  license: MIT
```
Comment (chrisburr) [inline recipes/bmi-fortran/meta.yaml]:
```suggestion
  license: MIT
  license_family: MIT
```

**Comment by mdpiper** [comment]
@chrisburr Thanks for the suggestions!

**Comment by scopatz** [comment]
Thanks @mdpiper

**Pull request merged**